src/modules/proto_dhcp/dhcpd.c

```diff
@@ -355,6 +355,15 @@ static int dhcp_process(REQUEST *request)
 		if (relay) relay->vp_ipaddr = vp->vp_ipaddr;
 	}
 
+	/*
+	 *	RFC 6842: If there's a DHCP-Client-Identifier ("uid") in the
+	 *	request then echo this in the reply.
+	 */
+	vp = fr_pair_find_by_num(request->packet->vps, 61, DHCP_MAGIC_VENDOR, TAG_ANY); /* DHCP-Client-Identifier */
+	if (vp) {
```

Comment:
I'd do:

`if (vp && !fr_pair_find_by_num(request->reply->vps, 61, DHCP_MAGIC_VENDOR, TAG_ANY))`

i.e. if it's in the packet and not in the reply, add it to the reply.

Reply:
That works. Pushed.